src/main/java/com/yelp/nrtsearch/server/luceneserver/highlights/HighlightFetchTask.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2022 Yelp Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yelp.nrtsearch.server.luceneserver.highlights;
+
+import static com.yelp.nrtsearch.server.luceneserver.highlights.HighlightSettingsHelper.createPerFieldSettings;
+
+import com.yelp.nrtsearch.server.grpc.Highlight;
+import com.yelp.nrtsearch.server.grpc.SearchResponse.Hit.Builder;
+import com.yelp.nrtsearch.server.grpc.SearchResponse.Hit.Highlights;
+import com.yelp.nrtsearch.server.luceneserver.IndexState;
+import com.yelp.nrtsearch.server.luceneserver.search.FetchTasks.FetchTask;
+import com.yelp.nrtsearch.server.luceneserver.search.SearchContext;
+import java.io.IOException;
+import java.util.Map;
+import org.apache.lucene.facet.taxonomy.SearcherTaxonomyManager.SearcherAndTaxonomy;
+import org.apache.lucene.index.IndexReader;
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.search.Query;
+
+public class HighlightFetchTask implements FetchTask {
```

Comment:
nit: would be good to have doc strings on public classes and methods, here and in other files